main.tf

```diff
@@ -41,25 +41,27 @@ resource "random_string" "deployment_id" {
 }
 
 resource "azurerm_resource_group" "aap" {
-  name     = var.resource_group
+  name = var.resource_group
   location = var.location
-  tags     = local.persistent_tags
+  tags = local.persistent_tags
 }
 
 resource "azurerm_virtual_network" "aap" {
-  name                = "${var.deployment_id}-aap-vn"
-  location            = azurerm_resource_group.aap.location
+  depends_on = [ azurerm_resource_group.aap ]
```

Comment:
Do we need this if there is `resource_group_name = azurerm_resource_group.aap.name`?  Or are we using it as a matter of ensuring the dependency no matter what?

Same comment on all the other depends_on additions

Reply:
I think you're right. If we are using `resource_group_name = azurerm_resource_group.aap.name` in resource creation then depends on not required. I'll update it here and for other too